### PR TITLE
polyfill URL and move storeConfig to separate file

### PR DESCRIPTION
**Pull request by marcaaron**
Just some light housekeeping. @KieranPringle noticed a test was failing. I think I tracked it down to the URL constructor @brortao likes to use :) Seems node doesn't have access to the `new URL()` syntax so I added a polyfill and the test now passes.

Oh and I also threw the redux middlewares into a separate file. Figured this might make passing the store around easier for testing purposes. But maybe there's a better way to handle this that I'm not aware of.

### CODE REVIEW

File: src/storeConfig.tsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { applyMiddleware, compose, createStore } from "redux";
+import thunk from "redux-thunk";
+
+import { rootReducer } from "./store";
+
+// enable Redux Devtools extension
+const composeEnhancers = (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
+const storeConfig = createStore(rootReducer, composeEnhancers(applyMiddleware(thunk)));
+export default storeConfig;
```
Comment (apodiptych) [inline src/storeConfig.tsx]: Since this exports a store and not a config object, can we call it something else? Maybe `reduxStore`?

Comment (marcaaron) [reply src/storeConfig.tsx]: yeah! good suggestion that makes sense :)

**Comment by nam20485** [comment]
The travis build is currently failing due to this unit test not passing. As soon as you merge this in, then the travis build should start succeeding.

**Comment by nam20485** [comment]
I am currently using `yarn test` to run the unit tests. Your webpack config specifies `yarn test` should  run `jest`. 

Does this sound correct? If not, all you need to change is what the `yarn test` command invokes and the travis pipeline will start running that once you commit it.

**Comment by marcaaron** [comment]
Yep ‘yarn test’ is correct! Just need a review on this one. :)

**Pull request merged**